app/lib/yahoo_finance.rb

```diff
@@ -35,15 +38,54 @@ def fetch_finance_hist(symbol)
       #   [5] - Volume
       #   [6] - Adjusted Close
 
-      days_past = 36500
+      days_past = 365
       hist = YahooFinance::get_historical_quotes_days(symbol, days_past)
       histJSON = hist.map { |e| {id: e}}.to_json
+
+      # Caching the historical data
+      cache(histJSON, symbol, true)
+
+      return histJSON
     end
 
     # Download and cache the current quote for target stock
     def fetch_finance_curr(symbol)
       # Get new data
       quote = YahooFinance::get_standard_quotes symbol
+      quoteJSON = quote[symbol].to_json
+      cache(quoteJSON, symbol, false)
+
+      return JSON.parse(quoteJSON)
+    end
+
+    # Stores the results in the DB
+    def cache(data, symbol, is_hist)
+      # :type is an enum which identifies historical (0) or current(1) data
+      data_type = is_hist ? :hist_data : :curr_data
+      cache_type = is_hist ? 0 : 1
+      yhoo_cache =
+        FinanceCache.new(data_type => data,
```

Comment:
data_type isnt a symbol, so not sure how to do the non rocket syntax with a variable symbol